### PR TITLE
refactor Group flow and some fixes around User

### PR DESCRIPTION
**Pull request by atayarani**
@Tapjoy/eng-group-ops 

Notes:
* Replaces #4

### CODE REVIEW

File: lib/tapjoy/ldap/group/delete.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
+module Tapjoy
+  module LDAP
+    module Group
+      # Delete LDAP group
+      class Delete
+        def delete
+          unless opts[:force] confirm
```
Comment (marcuswalser) [inline lib/tapjoy/ldap/group/delete.rb]: Getting this stacktrace when I run ldaptools `/Users/mwalser/.rbenv/versions/2.1.5/lib/ruby/gems/2.1.0/gems/ldap_tools-0.4.0/lib/tapjoy/ldap/group.rb:2:in `require_relative': /Users/mwalser/.rbenv/versions/2.1.5/lib/ruby/gems/2.1.0/gems/ldap_tools-0.4.0/lib/tapjoy/ldap/group/delete.rb:7: syntax error, unexpected tIDENTIFIER, expecting keyword_then or ';' or '\n' (SyntaxError)`, referencing this line.

Comment (atayarani) [inline lib/tapjoy/ldap/group/delete.rb]: I had the confirm unless word order wrong.  Please try again.

File: lib/tapjoy/ldap/group/delete.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+            opt :force, 'Force delete'
+          end
+        end
+
+        def dn
+          @dn ||= "cn=#{opts[:name]},ou=Group,#{Tapjoy::LDAP::client.basedn}"
+        end
+
+        def confirm
+          puts "Confirm that you want to delete group: #{ opts[:group] }"
```
Comment (marcuswalser) [inline lib/tapjoy/ldap/group/delete.rb]: I realize this is not a change but this prompt could be clearer about what acceptable inputs are - "Confirm that you want to delete group #{ opts[:group] } (yes/no)" or whatever.

**Comment by marcuswalser** [comment]
One small nit, functions normally for add/delete/add_user so LGTM.

**Pull request merged**